### PR TITLE
Adding support for fenceAgentRemediationTemplate

### PR DESCRIPTION
**Pull request by geetikakay**
##### Short description:
support for fenceAgentRemediationTemplate

### CODE REVIEW

File: ocp_resources/fence_agent_remediation_templates.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# API Reference: https://github.com/medik8s/fence-agents-remediation
```
Comment (omrirh) [inline ocp_resources/fence_agent_remediation_templates.py:1]: Please move to class docstring

**Comment by geetikakay** [comment]
@myakove Hi , pre-commit fails . That's how timeout should be : https://github.com/medik8s/fence-agents-remediation/blob/main/api/v1alpha1/fenceagentsremediation_types.go#L80

File: ocp_resources/fence_agent_remediation_templates.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    api_group = NamespacedResource.ApiGroup.FENCE_AGENTS_REMEDIATION_MEDIK8S_IO
+
+    def __init__(
+        self,
+        agent: str,
+        remediation_strategy: Optional[str] = None,
+        retry_count: Optional[int] = None,
+        retry_interval: Optional[str] = None,
+        timeout: Optional[str] = None,
```
Comment (myakove) [inline ocp_resources/fence_agent_remediation_templates.py:20]: Resource already have timeout, you need to set here different name.

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by myakove — DISMISSED** [review]

File: ocp_resources/fence_agent_remediation_templates.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    def to_dict(self) -> None:
+        super().to_dict()
+        if not self.yaml_file:
+            if not self.agent:
+                raise MissingRequiredArgumentError(argument="agent")
+            _spec = self.res["spec"] = {"template": {"spec": {}}}
+            _spec["template"]["spec"]["agent"] = self.agent
+            if self.remediation_strategy:
+                _spec["template"]["spec"]["remediationStrategy"] = self.remediation_strategy
+            if self.retry_count != 0:
```
Comment (rnetser) [inline ocp_resources/fence_agent_remediation_templates.py:55]:
```suggestion
            if self.retry_count is not None:
```

the API does value validation

File: ocp_resources/fence_agent_remediation_templates.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            shared_parameters (dict, optional): Parameters common to all nodes for the fencing agent.
+            node_parameters (dict, optional): Node-specific parameters for the fencing agent.
+        """
+        super().__init__(**kwargs)
+        self.agent = agent
+        self.remediation_strategy = remediation_strategy
+        self.retry_count = retry_count
+        self.retry_interval = retry_interval
+        self.remediation_timeout = remediation_timeout
+        self.shared_parameters = shared_parameters or {}
```
Comment (rnetser) [inline ocp_resources/fence_agent_remediation_templates.py:43]: no need to set to an empty dict

File: ocp_resources/fence_agent_remediation_templates.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        self.node_parameters = node_parameters or {}
```
Comment (rnetser) [inline ocp_resources/fence_agent_remediation_templates.py:44]: ditto

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/fence_agent_remediation_templates.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        remediation_timeout: Optional[str] = None,
+        shared_parameters: Optional[Dict[str, str]] = None,
+        node_parameters: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Initialize a FenceAgentsRemediationTemplate object.
+
+        Args:
+            agent (str): The name of the fence agent to use, must start with "fence_".
```
Comment (rnetser) [inline ocp_resources/fence_agent_remediation_templates.py:29]:
```suggestion
            agent (str): The name of the fence agent to use.
```

this is a constraint done by the API, not the wrapper code. please remove from docsring

**Review by rnetser — CHANGES_REQUESTED** [review]

**Review by rnetser — APPROVED** [review]

**Comment by geetikakay** [comment]
@myakove @rnetser Hello, can this be merged. Thanks

**Pull request merged**